app/assets/javascripts/admin/spree/orders/variant_autocomplete.js.erb

```diff
@@ -54,46 +54,53 @@ $(document).ready(function() {
     }
     $('a.save-item').click(handle_save_click);
 
-    handle_delete_click = function(){
-      var del = $(this);
-      var shipment_number = del.data('shipment-number');
-      var variant_id = del.data('variant-id');
+    handle_delete_click = function(e, confirmed){
+      if (confirmed) {
+        var del = $(this);
+        var shipment_number = del.data('shipment-number');
+        var variant_id = del.data('variant-id');
 
-      toggleItemEdit();
+        toggleItemEdit();
 
-      adjustItems(shipment_number, variant_id, 0);
+        adjustItems(shipment_number, variant_id, 0);
+      }
     }
-    $('a.delete-item').click(handle_delete_click);
+    $('a.delete-item').on('confirm:complete', handle_delete_click);
   }
 });
 
 adjustItems = function(shipment_number, variant_id, quantity){
-    var shipment = _.findWhere(shipments, {number: shipment_number + ''});
-    var inventory_units = _.where(shipment.inventory_units, {variant_id: variant_id});
-
-    var url = Spree.routes.orders_api + "/" + order_number + "/shipments/" + shipment_number;
-
-    var new_quantity = 0;
-    if(inventory_units.length<quantity){
-      url += "/add"
-      new_quantity = (quantity - inventory_units.length);
-    }else if(inventory_units.length>quantity){
-      url += "/remove"
-      new_quantity = (inventory_units.length - quantity);
-    }
-    url += '.json';
-
-    if (new_quantity == 0) {
-      alert(t("js.admin.orders.quantity_unchanged"));
-    } else {
-      $.ajax({
-        type: "PUT",
-        url: Spree.url(url),
-        data: { variant_id: variant_id, quantity: new_quantity }
-      }).done(function( msg ) {
-        window.location.reload();
-      });
-    }
+  var shipment = _.findWhere(shipments, {number: shipment_number + ''});
+  var inventory_units = _.where(shipment.inventory_units, {variant_id: variant_id});
+
+  if (quantity == 0 && inventory_units.length == shipment.inventory_units.length) {
+    alert(t("js.admin.orders.cannot_remove_last_item"));
```

Comment:
The issue mentions that "the warning Line items can't be blank should apply", which refers to a Rails flash error message (see Issue). However, I think it's easier and safer to validate this in the frontend (avoids interfering with other parts of the site), so I used an alert, as it being done elsewhere in this page. Is this ok?

Reply:
That seems fine to me - it's more descriptive this way as well 👍

Reply:
Does this `alert` looks like the default javascript alert modal?
Actually I don't know if it used somewhere else, and I'm not a huge fan of this default modal. 
I can see in the code a `info_dialog` maybe it's a good way to do this?